### ISSUE TITLE
Handle newlines generated by expression arguments

### ISSUE THREAD

**Issue by calculuschild** [adrianjost/dedent-tabs#165]
This is a great package! I just have run into one issue that I imagine could be fixed here.

If I have an expression argument that generates a substring, any newline characters in that substring disrupt the dedent process. For example:

```
return dedent`
	line 1
	line 2
	${generateText()}
	line 3
	line 4`;

const generateText = function(){
	return dedent`
		innertext 1
		innertext 2
		innertext 3`;
};
```

After `generateText()` runs, the string passed to `dedent` is :

```
	line 1
	line 2
	innertext 1
innertext 2
innertext 3`;
	line 3
	line 4
```

And naturally, dedenting fails here because two lines have no indent anymore. Would it be possible to insert a matching level of tabs in front of any newline characters found inside an expression? Probably by expanding line 24 here:

https://github.com/adrianjost/dedent-tabs/blob/97fe60b3b8b6520b704c6fc8a6f41fd90466835e/dedent.js#L23-L25

To just do a check if the preceding string matches `\n(\s+)$);`, i.e., see if the preceding string was an indent, and if so, apply that same indent in front of any other newlines found in `values[i]`;

What do you think? I can make a PR for this if it sounds reasonable.

**Comment by calculuschild** [comment]
Any chance of this getting looked at? Is this package still maintained?

**Comment by calculuschild** [comment]
Bumping again. I'd be happy to make a PR for this, but I'm not going to go through the effort if this project has been abandoned. Can we get any response from the owner?

@adrianjost

**Comment by adrianjost** [comment]
Hi @calculuschild I am very sorry for the late response. I am happy to look into any suggested improvements and a PR would be much appreciated. I think your initial idea would be a great improvement.

I just noticed that I must have a notification blind spot here so I didn't noticed this issue exists. Again: Sorry!

**Comment by calculuschild** [comment]
@adrianjost PR for ya.

**Comment by adrianjost** [comment]
Thanks for your contribution.
Your fix is now released in [v0.10.0](https://github.com/adrianjost/dedent-tabs/releases/tag/v0.10.0).
